### PR TITLE
[Docs]: Improve Events

### PR DESCRIPTION
**Pull request by johannesschobel**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/nestjs/docs.nestjs.com/blob/master/CONTRIBUTING.md

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[x] Docs
[ ] Other... Please describe:
```

## What is the current behavior?

This PR does the following thing:
- change the order of the sections in the `events` part. Now, first emitting events will be described, then listening to events
- the use of multilevel wildcards are explained. regular wildcards (`*`) are only one level deep!

Issue Number: N/A
https://github.com/nestjs/event-emitter/issues/25

## What is the new behavior?

Clarify the use of events and multilevel wildcards

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

**Comment by kamilmysliwiec** [comment]
@jmcdo29 can you give it a quick review? Thanks @johannesschobel!

File: content/techniques/events.md [hunk 1]
```diff
@@ -79,26 +101,16 @@ handleOrderEvents(payload: OrderCreatedEvent | OrderRemovedEvent | OrderUpdatedE
...
-
-To dispatch an event, first inject `EventEmitter2` using standard constructor injection:
-
-```typescript
-constructor(private eventEmitter: EventEmitter2) {}
-```
-
-> info **Hint** Import the `EventEmitter2` from the `@nestjs/event-emitter` package.
+Note that such a wildcard only applies to one block. The argument `order.*` will match, for example, the events `order.created` and `order.shipped` but not `order.delayed.out_of_stock`. In order to listen to such events,
+use the `multilevel wildcard` pattern (i.e, `**`), described in the `EventEmitter2` package.
```
Comment (jmcdo29) [inline content/techniques/events.md:105]: Any chance you can link to `EventEmitter2` documentation (GitHub, docs site, whatever they have)?

File: content/techniques/events.md [hunk 2]
```diff
@@ -50,7 +50,29 @@ EventEmitterModule.forRoot({
...
+
+To dispatch (i.e., fire) an event, first inject `EventEmitter2` using standard constructor injection:
+
+```typescript
+constructor(private eventEmitter: EventEmitter2) {}
+```
+
+> info **Hint** Import the `EventEmitter2` from the `@nestjs/event-emitter` package.
+
+Then use it in a class as follows.
```
Comment (jmcdo29) [inline content/techniques/events.md:63]:
```suggestion
Then use it in a class as follows:
```

**Review by jmcdo29 — CHANGES_REQUESTED** [review]
Overall, looks great. Made a few small comment that aren't necessary, but would be nice to fix up.

**Comment by johannesschobel** [comment]
@jmcdo29 : applied your minor change-requests
@kamilmysliwiec : all done, ready to merge

**Review by jmcdo29 — APPROVED** [review]
LGTM

File: content/techniques/events.md [hunk 3]
```diff
@@ -79,26 +101,16 @@ handleOrderEvents(payload: OrderCreatedEvent | OrderRemovedEvent | OrderUpdatedE
...
-this.eventEmitter.emit(
-  'order.created',
-  new OrderCreatedEvent({
-    orderId: 1,
-    payload: {},
-  }),
-);
+@OnEvent('**')
+handleEverything(payload: any) {
+  console.log('listener triggered');
```
Comment (kamilmysliwiec) [inline content/techniques/events.md:112]:
```suggestion
  // handle and process an event
```

**Comment by kamilmysliwiec** [comment]
Thanks!

**Pull request merged**